Helper/Data.php

```diff
@@ -524,6 +513,68 @@ private function getProductsRecords($storeId, $collection, $potentiallyDeletedPr
         ];
     }
 
+    /**
+     * Check if product can be index on Algolia
+     *
+     * @param Product $product
+     * @param int     $storeId
+     * @param boolean $throwExceptions
+     *
+     * @return boolean
+     *
+     * @throws AlgoliaProductDisabledException
+     * @throws AlgoliaProductDeletedException
+     * @throws AlgoliaProductNotVisibleException
+     * @throws AlgoliaProductOutOfStockException
+     */
+    public function productCanBeReindexed($product, $storeId, $throwExceptions = false)
```

Comment:
Can you rename the method to `canProductBeReindex`? It's a good practice that methods returning `bool` value starts with verb like: `is`, `has`, `should`, `can` ;)

Reply:
Parameter `$throwExceptions` can be omitted completely and only throw exceptions from here. And condition here can be replaced by:

```
try {
    /** @throws ... */
   $this->productCanBeReindexed($product, $storeId);
   // FIXME - prepare the product and add it to $productsToIndex
   $productsToIndex[$productId] = $this->productHelper->getObject($product);
} 
catch (AbstractAlgoliaProductException $e) {
    $productsToRemove[$productId] = $productId;
}

Would it make more sense?